common/src/test/java/io/honeycomb/opentelemetry/EnvironmentConfigurationTest.java

```diff
@@ -0,0 +1,171 @@
+    }
+
+    @Test
+    public void test_traces_apiendpoint() {
+        System.setProperty("honeycomb.traces.endpoint", "my-endpoint");
+        Assertions.assertEquals("my-endpoint", EnvironmentConfiguration.getHoneycombTracesApiEndpoint());
+    }
+
+    @Test
+    public void test_traces_apiendpoint_fallsback() {
+        System.setProperty("honeycomb.api.endpoint", "my-endpoint");
+        Assertions.assertEquals("my-endpoint", EnvironmentConfiguration.getHoneycombTracesApiEndpoint());
+    }
+
+    @Test
+    public void test_traces_dataset() {
+        System.setProperty("honeycomb.traces.dataset", "my-dataset");
+        Assertions.assertEquals("my-dataset", EnvironmentConfiguration.getHoneycombTracesDataset());
+    }
+
+    @Test
+    public void test_traces_dataset_fallsback() {
+        System.setProperty("honeycomb.dataset", "my-dataset");
+        Assertions.assertEquals("my-dataset", EnvironmentConfiguration.getHoneycombTracesDataset());
+    }
+
+    @Test
+    public void test_metrics_apikey() {
+        System.setProperty("honeycomb.metrics.apikey", "my-apikey");
+        Assertions.assertEquals("my-apikey", EnvironmentConfiguration.getHoneycombMetricsApiKey());
+    }
+
+    @Test
+    public void test_metrics_apikey_fallsback() {
+        System.setProperty("honeycomb.api.key", "my-apikey");
+        Assertions.assertEquals("my-apikey", EnvironmentConfiguration.getHoneycombMetricsApiKey());
+    }
+
+    @Test
+    public void test_metrics_apiendpoint() {
+        System.setProperty("honeycomb.metrics.endpoint", "my-apiendpoint");
+        Assertions.assertEquals("my-apiendpoint", EnvironmentConfiguration.getHoneycombMetricsApiEndpoint());
+    }
+
+    @Test
+    public void test_metrics_apiendpoint_fallsback() {
+        System.setProperty("honeycomb.api.endpoint", "my-apiendpoint");
+        Assertions.assertEquals("my-apiendpoint", EnvironmentConfiguration.getHoneycombMetricsApiEndpoint());
+    }
+
+    @Test
+    public void test_metrics_dataset() {
+        System.setProperty("honeycomb.metrics.dataset", "my-dataset");
+        Assertions.assertEquals("my-dataset", EnvironmentConfiguration.getHoneycombMetricsDataset());
+    }
+
+    @Test
+    public void test_metrics_dataset_does_not_fallback() {
+        System.setProperty("honeycomb.dataset", "my-dataset");
+        Assertions.assertEquals(null, EnvironmentConfiguration.getHoneycombMetricsDataset());
+    }
+
+    @Test
+    public void test_enableOtlpTraces_sets_system_properties() {
+        System.setProperty("honeycomb.api.key", "my-key");
+        System.setProperty("honeycomb.dataset", "my-dataset");
+
+        EnvironmentConfiguration.enableOTLPTraces();
+        Assertions.assertEquals("https://api.honeycomb.io:443", System.getProperty("otel.exporter.otlp.traces.endpoint"));
+        Assertions.assertEquals("X-Honeycomb-Team=my-key,X-Honeycomb-Dataset=my-dataset", System.getProperty("otel.exporter.otlp.traces.headers"));
```

Comment:
Do you think these assertions might be sufficient to address [the TODO item for testing that headers are set appropriately in the agent](https://github.com/honeycombio/honeycomb-opentelemetry-java/blob/e96452f6b95576cbb96d2505d490d3b912acf30d/sdk/src/test/java/io/honeycomb/opentelemetry/OpenTelemetryConfigurationTest.java#L55-L58)?